src/core/error/ApiError.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+
+import errorTypes from 'core/error/types';
+
+/**
+ * Errors sent by HTTP API / backend
+ */
+//eslint-disable-next-line
+export default class ApiError extends Error {
+    /**
+     * Instantiate an error
+     * @param {string} message- the error message
+     * @param {number} errorCode - the HTTP status or custom error code
```

Comment:
It looks fairly standard, but isn't this mixing 2 concepts? If TAO would one day have custom error codes, such as `E42 - bad QTI response format`, should they be stored in the same field which can be `500 - Server Error`?

Suggestion: keep `errorCode` and `httpStatusCode` apart.

Reply:
> Code review only, I'll test a bit later.
> 
> Question from [StackOverflow](https://stackoverflow.com/a/43595019): since we're compiling with Babel, do we need to have `babel-plugin-transform-builtin-extend` to make this work?

it looks like the current version of Babel handles it correctly, ie `instanceof` seems to be working as well as stack trace, but I didn't test on IE

Reply:
Ok then is the comment accurate for the intended usage? (`the HTTP status or custom error code`)